app/models/proof_attempt.rb

```diff
@@ -1,13 +1,35 @@
 class ProofAttempt < ActiveRecord::Base
+  include Numbering
+
+  numbering_parent_column 'sentence_id'
+
   belongs_to :theorem, foreign_key: 'sentence_id'
   belongs_to :proof_status
+  has_many :generated_axioms, dependent: :destroy
+  has_and_belongs_to_many :used_axioms,
+                          class_name: 'Sentence',
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.